zcash_client_sqlite/src/wallet/init/migrations/sent_notes_to_internal.rs

```diff
@@ -0,0 +1,88 @@
+//! A migration that adds an identifier for the account that received a sent note
+//! on an internal address to the sent_notes table.
+use std::collections::HashSet;
+
+use rusqlite;
+use schemer;
+use schemer_rusqlite::RusqliteMigration;
+use uuid::Uuid;
+
+use super::{addresses_table, utxos_table};
+use crate::wallet::init::WalletMigrationError;
+
+/// This migration adds the `to_account` field to the `sent_notes` table.
+///
+/// 0ddbe561-8259-4212-9ab7-66fdc4a74e1d
+pub(super) const MIGRATION_ID: Uuid = Uuid::from_fields(
+    0x0ddbe561,
+    0x8259,
+    0x4212,
+    b"\x9a\xb7\x66\xfd\xc4\xa7\x4e\x1d",
+);
+
+pub(super) struct Migration;
+
+impl schemer::Migration for Migration {
+    fn id(&self) -> Uuid {
+        MIGRATION_ID
+    }
+
+    fn dependencies(&self) -> HashSet<Uuid> {
+        [utxos_table::MIGRATION_ID, addresses_table::MIGRATION_ID]
+            .into_iter()
+            .collect()
+    }
+
+    fn description(&self) -> &'static str {
+        "Adds an identifier for the account that received an internal note to the sent_notes table"
+    }
+}
+
+impl RusqliteMigration for Migration {
+    type Error = WalletMigrationError;
+
+    fn up(&self, transaction: &rusqlite::Transaction) -> Result<(), WalletMigrationError> {
+        transaction.execute_batch("ALTER TABLE sent_notes ADD COLUMN to_account INTEGER;")?;
+
+        // `to_account` should be null for all migrated rows, since internal addresses
+        // have not been used for change or shielding prior to this migration.
```

Comment:
This comment is confusing. `to_account` is guaranteed to be null because we just added it, and we'd have hit a database error if the column already existed.

I also do not understand why we are altering the table to add a `to_account` column in the first place, given that we then immediately replace that table with a new one. We should instead check that the column does not exist in `sent_notes` (to replace the error that we'd get above) and then just use the new table creation.